dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -838,13 +458,10 @@ func TestCandidateValidation_validateFromChainState(t *testing.T) {
 		tt := tt
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
-			got, err := validateFromChainState(mockInstance, pov, tt.candidateReceipt)
-			if tt.expectedError != nil {
-				require.EqualError(t, err, tt.expectedError.Error())
-			} else {
-				require.NoError(t, err)
-			}
-			require.Equal(t, *tt.want, *got)
+
+			toSubsystem <- tt.msg
+			result := <-sender
```

Comment:
You are running all the test cases parallelly, So you should not share the channels among them.
You might receive the value from another case in the `sender` channel.

Reply:
I've updated this to make a new sender channel for each test run.